src/client/main.c

```diff
@@ -55,18 +77,33 @@ static int get_opts(int argc, char *argv[], char **opt_filter_glob) {
 
 int main(int argc, char *argv[]) {
         int r = 0;
-
-        _cleanup_free_ char *opt_filter_glob = NULL;
-        r = get_opts(argc, argv, &opt_filter_glob);
+        _cleanup_command_ Command *command = new_command();
+        r = get_opts(argc, argv, command);
         if (r < 0) {
                 return EXIT_FAILURE;
         }
-        if (no_action) {
-                return EXIT_SUCCESS;
+        if (command->op == NULL) {
+                usage();
+                return EXIT_FAILURE;
+        }
+        command->method = methods_get_method(command->op, methods);
+        if (command->method == NULL) {
+                if (command->op != NULL) {
```

Comment:
Extracting `if(command->is_help)` is incorrect, because we want to trigger usage here only when no command is given, otherwise we want to continue and trigger `command_execute` to later trigger command-specific help.

But, thanks to your comment I see that I have a bug in `get_opts`. Will refine this whole behavior

Reply:
Restructured the thing, now it works correctly